write-fonts/src/tables/gvar/iup.rs

```diff
@@ -1279,4 +1279,65 @@ mod tests {
             ]
         )
     }
+
+    // https://github.com/googlefonts/fontations/issues/662
+    #[test]
+    fn googlesans_m_sc() {
+        let deltas = vec![
+            Vec2 { x: 73.0, y: 0.0 },
+            Vec2 { x: 73.0, y: 580.0 },
+            Vec2 { x: 189.0, y: 580.0 },
+            Vec2 { x: 381.0, y: 116.0 },
+            Vec2 { x: 385.0, y: 116.0 },
+            Vec2 { x: 578.0, y: 580.0 },
+            Vec2 { x: 692.0, y: 580.0 },
+            Vec2 { x: 692.0, y: 0.0 },
+            Vec2 { x: 607.0, y: 0.0 },
+            Vec2 { x: 607.0, y: 342.0 },
+            Vec2 { x: 611.0, y: 448.0 },
+            Vec2 { x: 607.0, y: 448.0 },
+            Vec2 { x: 417.0, y: 0.0 },
+            Vec2 { x: 349.0, y: 0.0 },
+            Vec2 { x: 158.0, y: 448.0 },
+            Vec2 { x: 154.0, y: 448.0 },
+            Vec2 { x: 158.0, y: 342.0 },
+            Vec2 { x: 158.0, y: 0.0 },
+            Vec2 { x: 0.0, y: 0.0 },
+            Vec2 { x: 765.0, y: 0.0 },
+            Vec2 { x: 0.0, y: 0.0 },
+            Vec2 { x: 0.0, y: 0.0 },
+        ];
+
+        let coords = [
+            (73.0, 0.0),
+            (73.0, 580.0),
+            (189.0, 580.0),
+            (381.0, 116.0),
+            (385.0, 116.0),
+            (578.0, 580.0),
+            (692.0, 580.0),
+            (692.0, 0.0),
+            (607.0, 0.0),
+            (607.0, 342.0),
+            (611.0, 448.0),
+            (607.0, 448.0),
+            (417.0, 0.0),
+            (349.0, 0.0),
+            (158.0, 448.0),
+            (154.0, 448.0),
+            (158.0, 342.0),
+            (158.0, 0.0),
+            (0.0, 0.0),
+            (765.0, 0.0),
+            (0.0, 0.0),
+            (0.0, 0.0),
+        ]
+        .into_iter()
+        .map(|(x, y)| Point::new(x, y))
+        .collect::<Vec<_>>();
+
+        let contour_ends = vec![17];
+
+        iup_delta_optimize(deltas, coords, 0.5, &contour_ends).unwrap();
```

Comment:
can the test be further simplified? Do we need the entire outline to reproduce the problem?

(my intuition is it should simplify)